tests/v2/validation/observability/stackstate_test.go

```diff
@@ -0,0 +1,348 @@
+//go:build (validation || infra.any || cluster.any || sanity) && !stress && !extended
+
+package observability
+
+import (
+	"context"
+	"strings"
+
+	"testing"
+
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/rancher/tests/v2/actions/charts"
+	"github.com/rancher/rancher/tests/v2/actions/fleet"
+	"github.com/rancher/rancher/tests/v2/actions/kubeapi/namespaces"
+	kubeprojects "github.com/rancher/rancher/tests/v2/actions/kubeapi/projects"
+	"github.com/rancher/rancher/tests/v2/actions/observability"
+	"github.com/rancher/rancher/tests/v2/actions/projects"
+	"github.com/rancher/rancher/tests/v2/actions/uiplugins"
+	"github.com/rancher/shepherd/clients/rancher"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	extencharts "github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/clusters"
+	extensionscluster "github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/workloads/pods"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	project                 = "management.cattle.io.project"
+	rancherPartnerCharts    = "rancher-partner-charts"
+	systemProject           = "System"
+	localCluster            = "local"
+	stackStateConfigFileKey = "stackstateConfigs"
+	uiExtensionsRepo        = "https://github.com/rancher/ui-plugin-charts"
+	uiGitBranch             = "main"
+	rancherUIPlugins        = "rancher-ui-plugins"
+)
+
+type StackStateTestSuite struct {
+	suite.Suite
+	client                        *rancher.Client
+	session                       *session.Session
+	cluster                       *clusters.ClusterMeta
+	projectID                     string
+	stackstateAgentInstallOptions *charts.InstallOptions
+	stackstateConfigs             *observability.StackStateConfig
+}
+
+func (ss *StackStateTestSuite) TearDownSuite() {
+	ss.session.Cleanup()
+}
+
+func (ss *StackStateTestSuite) SetupSuite() {
```

Comment:
There are some differences between this SetupSuite and the rbac_test SetupSuite. For example, this one checks if the namespace already exists, and only creates if it doesn't. It does not throw an error if it exists. The rbac test tries to create the namespace each time, and throws an error if there was an error in that, which seems like then it would fail in the case where the namespace did already exist.

Is this intentional? Or should we have a separate stackstate.go file inside of this observability package with some functions to do some common things like in these Setups so as not to miss anything required between tests?

Reply:
good catch, thanks. rbac should be updated too incase the namespace exists and we are trying to create it each time the test would fail in the setup itself. let me fix that

Reply:
Do you think we should have a helper function in here so that both of these setups call the same code?

Reply:
Decided that we may create a helper in actions as part of @floatingman's work, depending on if it needs something similar. There may need to be multiple helpers split up, but we will see. Can refactor when necessary. For now it is okay.